src/java.desktop/share/classes/com/sun/media/sound/StandardMidiFileReader.java

```diff
@@ -146,34 +146,26 @@ private MidiFileFormat getMidiFileFormatFromStream(InputStream stream,
 
     @Override
     public MidiFileFormat getMidiFileFormat(URL url) throws InvalidMidiDataException, IOException {
-        InputStream urlStream = url.openStream(); // throws IOException
-        BufferedInputStream bis = new BufferedInputStream( urlStream, bisBufferSize );
-        MidiFileFormat fileFormat = null;
-        try {
-            fileFormat = getMidiFileFormat( bis ); // throws InvalidMidiDataException
-        } finally {
-            bis.close();
+        try (InputStream urlStream = url.openStream()) { // throws IOException
+            BufferedInputStream bis = new BufferedInputStream(urlStream, bisBufferSize);
```

Comment:
Do we need to close the bis here and below as well?

Reply:
I'm not sure it's 100% needed to close BufferedInputStream/BufferedReader wrappers. It's more code style question. Updated.